File: packages/shared/src/types/vault.ts
import type { VaultNoteId, WorkspaceId, SourceId } from "./ids.js";
import type { VaultNoteKind } from "./enums.js";

export interface VaultNoteFrontmatter {
  readonly title: string;
  readonly slug: string;
  readonly kind: VaultNoteKind;
  readonly workspaceId: WorkspaceId;
  readonly sourceIds: readonly SourceId[];
  readonly generatedAt: string;
  readonly model: string | null;
  readonly confidenceNotes: string | null;
  readonly backlinks: readonly string[];
  readonly tags: readonly string[];
}

export interface VaultNote {
  readonly id: VaultNoteId;
  readonly workspaceId: WorkspaceId;
  readonly kind: VaultNoteKind;
  readonly slug: string;
  readonly vaultPath: string;
  readonly frontmatter: VaultNoteFrontmatter;
  readonly contentHash: string;
  readonly createdAt: string;
  readonly updatedAt: string;
}

File: services/api/src/atlas_api/models/vault.py
"""Vault note domain models — mirrors packages/shared/src/types/vault.ts."""

from pydantic import BaseModel, ConfigDict

from atlas_api.models.enums import VaultNoteKind


class VaultNoteFrontmatter(BaseModel):
    model_config = ConfigDict(frozen=True)

    title: str
    slug: str
    kind: VaultNoteKind
    workspace_id: str
    source_ids: tuple[str, ...]
    generated_at: str
    model: str | None
    confidence_notes: str | None
    backlinks: tuple[str, ...]
    tags: tuple[str, ...]


class VaultNote(BaseModel):
    model_config = ConfigDict(frozen=True)

    id: str
    workspace_id: str
    kind: VaultNoteKind
    slug: str
    vault_path: str
    frontmatter: VaultNoteFrontmatter
    content_hash: str
    created_at: str
    updated_at: str
